test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/validators/NumberOfTraceIDsEqualToRequests.java

```diff
@@ -33,11 +41,25 @@ public NumberOfTraceIDsEqualToRequests() {}
      */
     @Override
     public List<MockSpanData> validate(List<MockSpanData> spans, int requests) {
-        Set<String> totalTraceIDs = spans.stream().map(MockSpanData::getTraceID).collect(Collectors.toSet());
+        Set<String> totalTraceIDs = spans.stream()
+            .filter(span -> isMatchingSpan(span))
+            .map(MockSpanData::getTraceID)
+            .collect(Collectors.toSet());
         List<MockSpanData> problematicSpans = new ArrayList<>();
         if (totalTraceIDs.size() != requests) {
             problematicSpans.addAll(spans);
         }
         return problematicSpans;
     }
+
+    private boolean isMatchingSpan(MockSpanData mockSpanData) {
+        if (mockSpanData.getAttributes().isEmpty()) {
```

Comment:
This seems not right, the condition should be over `attributes`: if `attributes` are empty, we always match

Reply:
My bad. Even I was thinking the same way :)